tests/test_session.py

```diff
@@ -34,9 +34,18 @@ def test_get_valid_source_file():
     # Session file from bmlab>=0.0.14
     assert get_valid_source(data_file_path('1D-y.session.h5'))\
            == data_file_path('1D-y.h5')
-    # Session file from bmlab>=0.0.14
-    assert get_valid_source(data_file_path('Unavailable.h5'))\
-           is None
+    # Non-existent source data file
+    with pytest.raises(FileNotFoundError):
```

Comment:
Can you do `with pytest.raises(FileNotFoundError) as err:` and then test whether `err.filename` is correctly set? This would be a test to ensure compatibility with impose (but also generally a good test).